src/detection/Footprint.cc

```diff
@@ -296,16 +296,16 @@ std::unique_ptr<Footprint> Footprint::readSpanSet(afw::table::BaseCatalog const&
 }
 
 void Footprint::readPeaks(afw::table::BaseCatalog const& peakCat, Footprint& loadedFootprint) {
-    if (!peakCat.getSchema().contains(PeakTable::makeMinimalSchema())) {
+    if (peakCat.getSchema().contains(PeakTable::makeMinimalSchema()) == 0) {
         // need to handle an older form of Peak persistence for backwards compatibility
         afw::table::SchemaMapper mapper(peakCat.getSchema());
         mapper.addMinimalSchema(PeakTable::makeMinimalSchema());
         afw::table::Key<float> oldX = peakCat.getSchema()["x"];
         afw::table::Key<float> oldY = peakCat.getSchema()["y"];
         afw::table::Key<float> oldPeakValue = peakCat.getSchema()["value"];
-        mapper.addMapping(oldX, "f.x");
-        mapper.addMapping(oldY, "f.y");
-        mapper.addMapping(oldPeakValue, "peakValue");
+        mapper.addMapping(oldX, true);
+        mapper.addMapping(oldY, true);
+        mapper.addMapping(oldPeakValue, true);
```

Comment:
This one really seems off...

Reply:
Well, could be. Or `clang-tidy` detected that it would always convert to `bool` and call that overload? We should check.

Reply:
Ah, crap, because a C-style string is actually a pointer. :man_facepalming: I don't suppose `clang-tidy` has a detector for ambiguous or surprising overload resolution?

I suppose we could try manually converting these to C++14 string literals, and seeing if that makes `clang-tidy` happy.

Reply:
Yeah, I believe that is what is happening. We expect it to call `Key<T> addMapping(Key<T> const& inputKey, std::string const& outputName, bool doReplace = true);`, but instead it calls `Key<T> addMapping(Key<T> const& inputKey, bool doReplace = false);` because it prefers to cast to `bool` rather than create a `std::string` temporary.